### ISSUE TITLE
Build fails because of Rustls error. 

### ISSUE THREAD

**Issue by SylvanFranklin** [avencera/rustywind#97]
While using `cargo install rustywind`, or `cargo build`

```sh
   #... 
   Compiling itertools v0.12.1
   Compiling colored v2.1.0
   Compiling indoc v2.0.4
   Compiling rustywind v0.21.0
error[E0308]: mismatched types
   --> /Users/sylvanfranklin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustywind-0.21.0/src/options/vite.rs:25:43
    |
25  |         agent = agent.tls_config(Arc::new(ssl));
    |                                  -------- ^^^ expected `rustls::client::client_conn::ClientConfig`, found `ClientConfig`
    |                                  |
    |                                  arguments to this function are incorrect
    |
    = note: `ClientConfig` and `rustls::client::client_conn::ClientConfig` have similar names, but are actually distinct types
note: `ClientConfig` is defined in crate `rustls`
   --> /Users/sylvanfranklin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustls-0.21.10/src/client/client_conn.rs:128:1
    |
128 | pub struct ClientConfig {
    | ^^^^^^^^^^^^^^^^^^^^^^^
note: `rustls::client::client_conn::ClientConfig` is defined in crate `rustls`
   --> /Users/sylvanfranklin/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustls-0.22.2/src/client/client_conn.rs:150:1
    |
150 | pub struct ClientConfig {
    | ^^^^^^^^^^^^^^^^^^^^^^^
    = note: perhaps two different versions of crate `rustls` are being used?
note: associated function defined here
   --> /Users/sylvanfranklin/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/sync.rs:386:12
    |
386 |     pub fn new(data: T) -> Arc<T> {
    |            ^^^

For more information about this error, try `rustc --explain E0308`.
error: could not compile `rustywind` (bin "rustywind") due to 1 previous error
error: failed to compile `rustywind v0.21.0`, intermediate artifacts can be found at `/var/folders/df/drlpsmlj3r99nb7t5mn3ymsm0000gn/T/cargo-installwEyfQ6`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```
I tried to change the version of Rustls, and got a slightly different error, but still something to do with client config. 

I'm on Cargo and Rust 1.76.0.

I installed fine with homebrew, but it would be nice to have closure on this.

**Comment by praveenperera** [comment]
Released new `v0.21.1` version that should fix it. Also try `cargo install rustywind --locked`

**Comment by SylvanFranklin** [comment]
Okay it works now!
